l10n_be_cooperator_portal/controllers/main.py

```diff
@@ -136,9 +131,8 @@ def portal_taxshelter_certificate(
             "subscription",
             "shares",
         ):
-            report_ref = (
-                "l10n_be_cooperator.action_tax_shelter_%s_report"
-                % (query_string)
+            report_ref = "l10n_be_cooperator.action_tax_shelter_%s_report" % (
```

Comment:
suggestion, non-blocking : `query_string` value can either be `subscrition` or `shares` : wouldn't it be clearer to rename `query_string` to a more explicit name (eg `report_type`)

AND/OR use explicit if/else such as

```py
  if query_string in == "subscription": 
      report_ref = "l10n_be_cooperator.action_tax_shelter_subscription_report"
  else:
      report_ref = "l10n_be_cooperator.action_tax_shelter_shares_report"
```

Reply:
very good point. renamed to `certificate_type`. i also used a common template for the print buttons to avoid duplication.